File: apps/backend/app/api/v1/ea_accounts.py
"""EA 服管账号管理路由（仅平台 admin 可访问）。

写入接收明文凭据并即时加密存储，读取只暴露健康状态，绝不回显任何明文凭据。
"""

from __future__ import annotations

from fastapi import APIRouter

from app.api.deps import CurrentUser, DbDep
from app.api.errors import ForbiddenError
from app.models import User
from app.schemas.ea_account import (
    EAAccountCreate,
    EAAccountCredentialsUpdate,
    EAAccountDisplayNameUpdate,
    EAAccountEnabledUpdate,
    EAAccountItem,
    EAAccountVerifyResult,
)
from app.services.ea_account_service import EAAccountService

router = APIRouter()


def _require_admin(user: User) -> None:
    if user.role != "admin":
        raise ForbiddenError(message="仅平台管理员可管理 EA 账号")


@router.get("", response_model=list[EAAccountItem])
async def list_ea_accounts(db: DbDep, user: CurrentUser) -> list[EAAccountItem]:
    _require_admin(user)
    return await EAAccountService(db).list_all()


@router.post("", response_model=EAAccountItem, status_code=201)
async def create_ea_account(
    payload: EAAccountCreate,
    db: DbDep,
    user: CurrentUser,
) -> EAAccountItem:
    _require_admin(user)
    return await EAAccountService(db).create(payload)


@router.patch("/{account_id}/credentials", response_model=EAAccountItem)
async def update_ea_account_credentials(
    account_id: int,
    payload: EAAccountCredentialsUpdate,
    db: DbDep,
    user: CurrentUser,
) -> EAAccountItem:
    _require_admin(user)
    return await EAAccountService(db).update_credentials(account_id, payload)


@router.patch("/{account_id}/enabled", response_model=EAAccountItem)
async def set_ea_account_enabled(
    account_id: int,
    payload: EAAccountEnabledUpdate,
    db: DbDep,
    user: CurrentUser,
) -> EAAccountItem:
    _require_admin(user)
    return await EAAccountService(db).set_enabled(account_id, payload.enabled)


@router.patch("/{account_id}/display-name", response_model=EAAccountItem)
async def update_ea_account_display_name(
    account_id: int,
    payload: EAAccountDisplayNameUpdate,
    db: DbDep,
    user: CurrentUser,
) -> EAAccountItem:
    _require_admin(user)
    return await EAAccountService(db).update_display_name(account_id, payload)


@router.delete("/{account_id}", status_code=204)
async def delete_ea_account(account_id: int, db: DbDep, user: CurrentUser) -> None:
    _require_admin(user)
    await EAAccountService(db).delete(account_id)


@router.post("/{account_id}/verify", response_model=EAAccountVerifyResult)
async def verify_ea_account(
    account_id: int,
    db: DbDep,
    user: CurrentUser,
) -> EAAccountVerifyResult:
    _require_admin(user)
    return await EAAccountService(db).verify(account_id)

File: apps/web/lib/api/ea-accounts.ts
import { api } from "@/lib/api-client";

export interface EAAccountItem {
  id: number;
  persona_id: number;
  display_name: string | null;
  enabled: boolean;
  last_used_at: string | null;
  failure_count: number;
  use_count: number;
  has_session: boolean;
  has_access_token: boolean;
  created_at: string;
  updated_at: string;
}

export interface EAAccountCreateRequest {
  persona_id: number;
  display_name?: string | null;
  remid: string;
  sid: string;
  session?: string | null;
  access_token?: string | null;
  enabled?: boolean;
}

export interface EAAccountCredentialsUpdateRequest {
  remid?: string | null;
  sid?: string | null;
  session?: string | null;
  access_token?: string | null;
}

export interface EAAccountDisplayNameUpdateRequest {
  /** 传 null 显式清空备注；传字符串覆盖。 */
  display_name: string | null;
}

export interface EAAccountVerifyResult {
  success: boolean;
  persona_id: number;
  message: string | null;
}

export const eaAccountsApi = {
  list: () => api.get<EAAccountItem[]>("/ea-accounts"),

  create: (payload: EAAccountCreateRequest) => api.post<EAAccountItem>("/ea-accounts", payload),

  updateCredentials: (id: number, payload: EAAccountCredentialsUpdateRequest) =>
    api.patch<EAAccountItem>(`/ea-accounts/${id}/credentials`, payload),

  updateDisplayName: (id: number, payload: EAAccountDisplayNameUpdateRequest) =>
    api.patch<EAAccountItem>(`/ea-accounts/${id}/display-name`, payload),

  setEnabled: (id: number, enabled: boolean) =>
    api.patch<EAAccountItem>(`/ea-accounts/${id}/enabled`, { enabled }),

  delete: (id: number) => api.delete<void>(`/ea-accounts/${id}`),

  verify: (id: number) => api.post<EAAccountVerifyResult>(`/ea-accounts/${id}/verify`),
};
